source/audio/utils.py

```diff
@@ -57,15 +65,31 @@ def unregister(self):
 			log.exception(f"Cannot unregister audio session for process {self.pid}")
 
 
-class AudioSessionNotificationListener(AudioSessionNotification):
+class _AudioSessionNotificationListener(AudioSessionNotification):
+	"""
+		This class is a handler for existing and newly created audio sessions. Its method `on_session_created`
+		will be called from `AudioSessionCallback.register()` for all existing audio sessions; and, additionally,
+		it will be notified by WASAPI for every new audio session created.
+		It sets up a callback for session termination - an instance of `_AudioSessionEventsListener` class;
+		then it calls custom logic provided in `AudioSessionCallback.onSessionUpdate()` implementation.
+	"""
 	callback: "weakref.ReferenceType[AudioSessionCallback]"
 
 	def __init__(self, callback: "AudioSessionCallback"):
 		self.callback = weakref.ref(callback)
 
 	def on_session_created(self, new_session: AudioSession):
 		pid = new_session.ProcessId
-		audioSessionEventsListener = AudioSessionEventsListener(self.callback(), pid, new_session)
+		if pid != globalVars.appPid:
+			process = new_session.Process
+			if process is not None:
+				exe = os.path.basename(process.exe())
+				isNvda = exe.lower() == "nvda.exe"
+				if isNvda:
+					# This process must be NVDA running on secure screen.
+					# We shouldn't change volume of such process.
+					return

```

Comment:
shouldn't this just use `isRunningOnSecureDesktop`

Reply:
No, here we're checking another process called nvda.exe and assuming that must be NVDA running on secure screen, therefore we shouldn't fidget  with its volume. On the other side `isRunningOnSecureDesktop` tells whether our process is running on secure desktop.

Reply:
Seems however that you cannot differentiate between the NVDA running on secure screen and an NVDA run in the session of another user. Am I correct?

Reply:
Correct.

Reply:
You can use `post_secureDesktopStateChange` for the other side of things, by checking if we have switched to the secure desktop from a user instance of NVDA.

Reply:
You can use post_secureDesktopStateChange for the other side of things, by checking if we have switched to the secure desktop from a user instance of NVDA.